packaging/package_info.rst

```diff
@@ -0,0 +1,190 @@
+
+Read more about this in :ref:`how_to_define_abi_compatibility`
+
+build_id
+++++++++++
+The ``build_id()`` methods is an optimization. If you find that you are doing exactly the same
+build for two different packages, then you might want to use this method to redefine the build ID.
+The build ID, by default is the same as the package ID, i.e. there is one build folder per package.
+But if for any reason you have a build system that is building different artifacts in the same
+build, and you want to create different packages with those artifacts, depending on different
+settings, you don't want to rebuild again the same, as it it usually time consuming.
+
+Read more about this in :ref:`build_id`
+
+Package information
+---------------------
+
+Single configuration
++++++++++++++++++++++
+
+A typical approach is to have each package contain the artifacts just for one configuration. In this
+approach, for example, the debug pre-compiled libraries will be in a different package than the
+release pre-compiled libraries.
+
+In this approach, the ``package_info()`` method can just set the appropriate values for consumers,
+to let them know about the package library names, and necessary definitions and compile flags.
+
+.. code-block:: python
+  
+   def package_info(self):
+        self.cpp_info.libs = ["mylib"]
+        
+The values that packages declare here (the ``include``, ``lib`` and ``bin`` subfolders are already
+defined by default, so they define the include and library path to the package) are translated
+to variables of the respective build system by the used generators. That is, if using the ``cmake``
+generator, such above definition will be translated in ``conanbuildinfo.cmake`` to something like:
+
+.. code-block:: cmake
+  
+    set(CONAN_LIBS_MYPKG mylib)
+    ...
+    set(CONAN_LIBS mylib ${CONAN_LIBS})
+    
+Those variables, will be used in the ``conan_basic_setup()`` macro to actually set cmake
+relevant variables.
+
+        
+Read more about this in :ref:`package_info`
+
+Multi configuration
++++++++++++++++++++++
+
+It is possible that someone wants to package both debug and release artifacts in the same package,
+so it can be consumed from IDEs like Visual Studio changing debug/release configuration from the IDE,
+and not having to specify it in the command line.
+
+Creating a multi-configuration Debug/Release package is not difficult, using ``CMake`` for example
+could be:
+
+
+.. code-block:: python
+
+    def build(self):
+        cmake = CMake(self.settings)
+        if cmake.is_multi_configuration:
+            cmd = 'cmake "%s" %s' % (self.conanfile_directory, cmake.command_line)
+            self.run(cmd)
+            self.run("cmake --build . --config Debug")
+            self.run("cmake --build . --config Release")
+        else:
+            for config in ("Debug", "Release"):
+                self.output.info("Building %s" % config)
+                self.run('cmake "%s" %s -DCMAKE_BUILD_TYPE=%s'
```

Comment:
Not really. The helpers use the standard ``cmake.command_line`` which depend on `sèttings.build_type`` in this case not defined. So I am manually overriding the behavior.  Probably it could be wrapped in another cmake helper method.